addons/account/models/account_invoice.py

```diff
@@ -122,6 +122,29 @@ def _compute_residual(self):
             self.reconciled = True
         else:
             self.reconciled = False
+    
+    @api.multi
+    def _get_domain_match_to_check(self):
+        self.ensure_one()
+        domain = [
+            ('move_id.to_check', '=', True),
+            ('full_reconcile_id', '=', False),
+        ]
+        if self.type == 'out_invoice':
+            domain.append(('balance', '=', -self.residual))
+        elif self.type == 'in_invoice':
+            domain.append(('balance', '=', self.residual))
+        elif self.type == 'out_refund':
+            domain.append(('balance', '=', self.residual))
+        elif self.type == 'in_refund':
+            domain.append(('balance', '=', -self.residual))
```

Comment:
as in action_reconcile_to_check() you're doing `mapped('statement_line_id').ids`, I think you need to this to the domain as well, otherwise it just gonna be weird as you may end up seeing the message that won't be opening any statement line in the reconciliation widget

Reply:
@avw asked to remove the constraint and changed the specs considering what we should expect as an expense account